### PR TITLE
PIL-971 Saving non uk upe/fm safe ids

### PR DESCRIPTION
**Pull request by mrah93**

### CODE REVIEW

File: app/pages/UpeNonUKSafeIDPage.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages
+
+import play.api.libs.json.JsPath
+
+case object UpeNonUKSafeIDPage extends QuestionPage[String] {
```
Comment (jassalrichy) [inline app/pages/UpeNonUKSafeIDPage.scala:21]: Add a page spec.

File: app/pages/FmNonUKSafeIDPage.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages
+
+import play.api.libs.json.JsPath
+
+case object FmNonUKSafeIDPage extends QuestionPage[String] {
```
Comment (jassalrichy) [inline app/pages/FmNonUKSafeIDPage.scala:21]: Add a page spec.

Comment (mrah93) [reply app/pages/FmNonUKSafeIDPage.scala:21]: Done

File: app/pages/UpeNonUKSafeIDPage.scala (hunk 1 above)
Comment (mrah93) [reply app/pages/UpeNonUKSafeIDPage.scala:21]: Done

File: app/services/SubscriptionService.scala [hunk 3]
```diff
@@ -42,12 +42,14 @@ class SubscriptionService @Inject() (
 
   def createSubscription(userAnswers: UserAnswers)(implicit hc: HeaderCarrier): Future[String] =
     for {
-      upeSafeId       <- registerUpe(userAnswers)
-      fmnSafeId       <- registerNfm(userAnswers)
+      upeSafeId     <- registerUpe(userAnswers)
+      latestAnswers <- userAnswersConnectors.getUserAnswer(userAnswers.id)
+      updatedAnswers = latestAnswers.getOrElse(userAnswers)
+      fmnSafeId       <- registerNfm(updatedAnswers)
```
Comment (jassalrichy) [inline app/services/SubscriptionService.scala:48]: rename fmnSafeId to nfmSafeId

Comment (mrah93) [reply app/services/SubscriptionService.scala]: Done

**Review by jassalrichy — APPROVED** [review]

**Pull request merged**